fplus-lib/src/core/mod.rs

```diff
@@ -3018,10 +3026,19 @@ impl LDNApplication {
     async fn is_allowance_sufficient(
         address: &str,
         new_allocation_amount: &str,
+        contract_address: Option<String>,
     ) -> Result<(), LDNError> {
-        let allowance = get_allowance_for_address(address)
-            .await
-            .map_err(|e| LDNError::Load(format!("Failed to retrieve allowance: {}", e)))?;
+        let allowance;
+
+        if let Some(contract_address) = contract_address {
+            allowance = get_allowance_for_address_contract(address, &contract_address)
```

Comment:
In that case shouldn't we also check how much DC the contract itself has and choose the lower number?